### PR TITLE
Test(eos_designs): Add test for rfc5549 with `overlay_routing_protocol: none`

### LINKED ISSUE #3522: Test(eos_designs): Add tests for rfc5549, overlay_routing_protocol: none for p2p-vrfs uplink
Add tests for rfc5549, overlay_routing_protocol: none with bgp peers etc under network services

_Originally posted by @gmuloc in https://github.com/aristanetworks/avd/pull/3467#discussion_r1461598478_

### PR DESCRIPTION
**Pull request by laxmikantchintakindi**
## Related Issue(s)

Fixes #3522

## Component(s) name

`arista.avd.eos_designs`

## Proposed changes
Add tests for rfc5549 with overlay routing protocol none.

### User Checklist

- N/A

### Repository Checklist

- [x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://avd.arista.com/stable/docs/contribution/overview.html) document.
- [ ] My change requires a change to the documentation and documentation have been updated accordingly.
- [x] I have updated [molecule CI](https://github.com/aristanetworks/avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly. (check the box if not applicable)

### CODE REVIEW

File: ansible_collections/arista/avd/molecule/eos_designs_unit_tests/inventory/hosts.yml [hunk 1]
```diff
@@ -462,6 +462,13 @@ all:
             UPLINK_P2P_VRFS_TESTS_SPINE2:
             UPLINK_P2P_VRFS_TESTS_LEAF1:
             UPLINK_P2P_VRFS_TESTS_L2LEAF1:
+        UPLINK_P2P_VRFS_RFC5549_TESTS:
+          hosts:
+            UPLINK_P2P_VRFS_RFC5549_TESTS_SPINE1:
```
Comment (gmuloc) [inline ansible_collections/arista/avd/molecule/eos_designs_unit_tests/inventory/hosts.yml:498]: check if we can reduce the number of devices

Comment (laxmikantchintakindi) [reply ansible_collections/arista/avd/molecule/eos_designs_unit_tests/inventory/hosts.yml:498]: Done.

File: ansible_collections/arista/avd/molecule/eos_designs_unit_tests/inventory/host_vars/UPLINK_P2P_VRFS_RFC5549_TESTS_LEAF.yml [hunk 2]
```diff
@@ -0,0 +1,2 @@
+---
+overlay_routing_protocol: none
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/molecule/eos_designs_unit_tests/inventory/host_vars/UPLINK_P2P_VRFS_RFC5549_TESTS_LEAF.yml]:
```suggestion
# Testing RFC5549 with IPv6 overlay peers only
overlay_routing_protocol: none
```

File: ansible_collections/arista/avd/molecule/eos_designs_unit_tests/intended/configs/UPLINK_P2P_VRFS_RFC5549_TESTS_LEAF.cfg [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+!
+router bgp 65000
+   router-id 192.168.42.8
+   update wait-install
+   no bgp default ipv4-unicast
+   maximum-paths 4 ecmp 4
+   neighbor IPv4-UNDERLAY-PEERS peer group
+   neighbor IPv4-UNDERLAY-PEERS send-community
+   neighbor IPv4-UNDERLAY-PEERS maximum-routes 12000
+   neighbor 10.42.42.28 peer group IPv4-UNDERLAY-PEERS
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/molecule/eos_designs_unit_tests/intended/configs/UPLINK_P2P_VRFS_RFC5549_TESTS_LEAF.cfg]: I don't see any RFC5549 in here, so are you sure you are testing what you think you are?

**Review by ClausHolbechArista — CHANGES_REQUESTED** [review]

File: ansible_collections/arista/avd/molecule/eos_designs_unit_tests/intended/configs/UPLINK_P2P_VRFS_RFC5549_TESTS_LEAF.cfg (hunk 3 above)
Comment (laxmikantchintakindi) [reply ansible_collections/arista/avd/molecule/eos_designs_unit_tests/intended/configs/UPLINK_P2P_VRFS_RFC5549_TESTS_LEAF.cfg]: Added underlay_rfc5549: true.

File: ansible_collections/arista/avd/molecule/eos_designs_unit_tests/inventory/group_vars/UPLINK_P2P_VRFS_RFC5549_TESTS.yml [hunk 4]
```diff
@@ -0,0 +1,84 @@
+---
+# Test uplink types p2p-vrfs with rfc5549 and 'overlay_routing_protocol: none'
+overlay_routing_protocol: none
+
+underlay_ipv6: true
+underlay_rfc5549: true
+
+node_type_keys:
+  # modify spine to get l3 network_services
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/molecule/eos_designs_unit_tests/inventory/group_vars/UPLINK_P2P_VRFS_RFC5549_TESTS.yml]: Just use l3spine instead

Comment (ClausHolbechArista) [reply ansible_collections/arista/avd/molecule/eos_designs_unit_tests/inventory/group_vars/UPLINK_P2P_VRFS_RFC5549_TESTS.yml]: not sure why we need services on the spine though

File: ansible_collections/arista/avd/molecule/eos_designs_unit_tests/inventory/group_vars/UPLINK_P2P_VRFS_RFC5549_TESTS.yml [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+    virtual_router_mac_address: cafe:cafe:cafe
+    uplink_ipv4_pool: 10.42.42.0/24
+    uplink_switches: [ UPLINK_P2P_VRFS_RFC5549_TESTS_SPINE1]
+    uplink_interfaces: [ Ethernet1, Ethernet2 ]
+    uplink_type: p2p-vrfs
+    uplink_macsec:
+      profile: TEST
+  nodes:
+    - name: UPLINK_P2P_VRFS_RFC5549_TESTS_LEAF
+      # Testing RFC5549 with IPv6 overlay peers only
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/molecule/eos_designs_unit_tests/inventory/group_vars/UPLINK_P2P_VRFS_RFC5549_TESTS.yml]: What do you mean here?

File: ansible_collections/arista/avd/molecule/eos_designs_unit_tests/inventory/group_vars/UPLINK_P2P_VRFS_RFC5549_TESTS.yml [hunk 6]
```diff
@@ -0,0 +1,84 @@
+---
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/molecule/eos_designs_unit_tests/inventory/group_vars/UPLINK_P2P_VRFS_RFC5549_TESTS.yml:1]: It seems like you have copied vars from something else. Please try to clean up to only have the relevant vars here

File: ansible_collections/arista/avd/molecule/eos_designs_unit_tests/inventory/group_vars/UPLINK_P2P_VRFS_RFC5549_TESTS.yml (hunk 4 above)
Comment (laxmikantchintakindi) [reply ansible_collections/arista/avd/molecule/eos_designs_unit_tests/inventory/group_vars/UPLINK_P2P_VRFS_RFC5549_TESTS.yml]: Removed node_type_keys setting.

File: ansible_collections/arista/avd/molecule/eos_designs_unit_tests/inventory/group_vars/UPLINK_P2P_VRFS_RFC5549_TESTS.yml (hunk 5 above)
Comment (laxmikantchintakindi) [reply ansible_collections/arista/avd/molecule/eos_designs_unit_tests/inventory/group_vars/UPLINK_P2P_VRFS_RFC5549_TESTS.yml]: Removed.

File: ansible_collections/arista/avd/molecule/eos_designs_unit_tests/inventory/group_vars/UPLINK_P2P_VRFS_RFC5549_TESTS.yml (hunk 6 above)
Comment (laxmikantchintakindi) [reply ansible_collections/arista/avd/molecule/eos_designs_unit_tests/inventory/group_vars/UPLINK_P2P_VRFS_RFC5549_TESTS.yml:1]: My apologies. Fixed.

File: ansible_collections/arista/avd/molecule/eos_designs_unit_tests/inventory/group_vars/UPLINK_P2P_VRFS_RFC5549_TESTS.yml [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+    platform: vEOS-LAB
+    loopback_ipv4_pool: 192.168.42.0/24
+    vtep_loopback_ipv4_pool: 172.16.0.0/24
+    virtual_router_mac_address: cafe:cafe:cafe
+    uplink_ipv4_pool: 10.42.42.0/24
+    uplink_switches: [ UPLINK_P2P_VRFS_RFC5549_TESTS_SPINE1]
+    uplink_interfaces: [ Ethernet1, Ethernet2 ]
+    uplink_type: p2p-vrfs
+    filter:
+      always_include_vrfs_in_tenants: [TenantA]
```
Comment (gmuloc) [inline ansible_collections/arista/avd/molecule/eos_designs_unit_tests/inventory/group_vars/UPLINK_P2P_VRFS_RFC5549_TESTS.yml:31]: please add this on the SPINE as well as the goal of the issue is to test p2p-vrfs on the uplink being configured.

Can you also add a bgp peer in one VRF as described in the issue?

**Review by gmuloc — CHANGES_REQUESTED** [review]

File: ansible_collections/arista/avd/molecule/eos_designs_unit_tests/inventory/group_vars/UPLINK_P2P_VRFS_RFC5549_TESTS.yml (hunk 7 above)
Comment (laxmikantchintakindi) [reply ansible_collections/arista/avd/molecule/eos_designs_unit_tests/inventory/group_vars/UPLINK_P2P_VRFS_RFC5549_TESTS.yml:31]: Done.

File: ansible_collections/arista/avd/molecule/eos_designs_unit_tests/intended/configs/UPLINK_P2P_VRFS_RFC5549_TESTS_SPINE1.cfg [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+vrf instance MGMT
+!
+management api http-commands
+   protocol https
+   no shutdown
+   !
+   vrf MGMT
+      no shutdown
+!
+interface Ethernet3
```
Comment (gmuloc) [inline ansible_collections/arista/avd/molecule/eos_designs_unit_tests/intended/configs/UPLINK_P2P_VRFS_RFC5549_TESTS_SPINE1.cfg:26]: I would have expected a subinterface created for the VRF IT here the way p2p-vrfs works (and actually some for the PROD one as well)

can you investigate why they are not created? 
Thanks!

Comment (gmuloc) [reply ansible_collections/arista/avd/molecule/eos_designs_unit_tests/intended/configs/UPLINK_P2P_VRFS_RFC5549_TESTS_SPINE1.cfg:26]: updated test in latest commit

**Review by gmuloc — APPROVED** [review]

**Review by ClausHolbechArista — APPROVED** [review]
